contents/blog/data-warehouse-at-posthog.mdx

```diff
@@ -0,0 +1,353 @@
+import PipelinePerQuarter from './_snippets/pipeline-per-quarter.mdx'
+import ManagedAccounts from './_snippets/managed-accounts.mdx'
+import { CalloutBox } from 'components/Docs/CalloutBox'
+
+PostHog's [data warehouse](/data-warehouse) is our most powerful feature. It lets you sync data from the tools you already use like [Stripe](/docs/cdp/sources/stripe), [Salesforce](/docs/cdp/sources/salesforce), and [Hubspot](/docs/cdp/sources/hubspot), query it alongside your existing product data using [SQL](/docs/data-warehouse/sql), and visualize it natively.
+
+We built it because the [modern data stack sucks](/blog/modern-data-stack-sucks). What starts as a handful of business critical tools devolves into dozens of tools, many specifically built to capture, clean, format, load, query, and visualize data.
+
+We knew it didn't have to be this way, so we built the data warehouse to get rid of all this complexity and give you a single source of truth for all your business data.
+
+<ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2025_07_16_at_11_04_28_2x_905d242fe0.png"
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2025_07_16_at_11_04_02_2x_690da5bd8b.png"
+  alt="PostHog data warehouse rows synced"
+  classes="rounded"
+/>
+<Caption>We sync hundreds of millions of rows from Postgres, Stripe, Temporal, and more on top of the tens of millions of events we capture each day.</Caption>
+
+We've created **over 1,600** SQL insights and visualizations using our data warehouse so far. It's our second most-used insight type behind [trends](/docs/product-analytics/trends/overview), which was around long before we had the data warehouse.
+
+<ProductScreenshot
+  imageLight="https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2025_07_10_at_11_47_15_2x_9b179c988a.png"
+  imageDark="https://res.cloudinary.com/dmukukwp6/image/upload/Clean_Shot_2025_07_10_at_11_47_58_2x_19e2267c12.png"
+  alt="PostHog SQL insights"
+  classes="rounded"
+/>
+
+To help you get started, we're sharing how teams at PostHog use our built-in data warehouse and custom SQL insights to answer critical business questions, like:
+
+- Which customers churned and how did it impact revenue?
+- Which customers are increasing their spend?
+- Who are our biggest customers and what products are they using?
+- Are we achieving our customer support goals?
+- What are the biggest sales opportunities in our pipeline?
+
+We're also sharing the actual SQL queries we use to answer these questions and the dashboards we use to visualize the data.
+
+<CalloutBox icon="IconInfo" title="Modifying queries to your data" type="fyi">
+  Because our data structure is unique, the SQL queries included here likely won't work out-of-the-box for you. Luckily, [Max AI](/max) makes modifying SQL easy. Just paste these queries in as context and ask him to use your data instead.
```

Comment:
```suggestion
  Because our data structure is unique, the SQL queries included here likely won't work out-of-the-box for you. Luckily, <a href="/max">Max AI</a> makes modifying SQL easy. Just paste these queries in as context and ask him to use your data instead.
```